### PR TITLE
#781 - Search by product variations SKU

### PR DESCRIPTION
**Pull request by lukaspawlik**
@tlovett1 This is a proof of concept of handling searching by products variations SKU and returning parent product (https://github.com/10up/ElasticPress/issues/781).

This is quite experimental approach as it seems that the easiest way of achieving this is creating artificial meta with SKUs of _product_variation_ children posts. Values of this artificial meta is calculated during posts indexing and it supports scenarios when variation product is edited/added/removed.

If you don't like using _meta_ for this then how about creating extra post property for storing this kind of "dynamically calculated static data" separately from metas?

### CODE REVIEW

**Comment by kallehauge** [comment]
It is AWESOME that you've started this work @lukaspawlik !

Feedback: the logic for syncing a parent when a variation is changed is of course reusable – but if I wish to add variation descriptions or a custom field to be searchable for the parent, then I would have to re-implement the rest of the logic (I don't feel like this should, in any way, be a show-stopper for release 2.4 but maybe a feature-request for 2.5?)

**Comment by kallehauge** [comment]
@lukaspawlik have you, internally, discussed this approach? And is there anything I can do to make this go faster? From what I have gathered from some of my clients, then this is the last feature they feel like they need to have the WooCommerce functionalities they had before

**Comment by kallehauge** [comment]
@lukaspawlik do you know if this will be in the next release of the plugin? Let me know if I can do anything and I'll try to find some time

**Comment by lukaspawlik** [comment]
@kallehauge I need to defer this to @eugene-manuilov as he can have more reliable information than I have. @eugene-manuilov could you please take over this ticket?

File: features/woocommerce/woocommerce.php [hunk 1]
```diff
@@ -531,6 +531,70 @@ function ep_wc_remove_legacy_meta( $post_args, $post_id ) {
...
+ *
+ * @return array
+ */
+function ep_wc_add_variations_skus( $post_args, $post_id ) {
+	global $wpdb;
+	if ( 'product' !== get_post_type( $post_id ) ) {
+		return $post_args;
+	}
+	$skus = $wpdb->get_results(
+		$wpdb->prepare( "SELECT {$wpdb->posts}.ID, {$wpdb->postmeta}.meta_value FROM {$wpdb->posts} LEFT JOIN {$wpdb->postmeta} ON {$wpdb->posts}.ID = {$wpdb->postmeta}.post_id  WHERE {$wpdb->postmeta}.meta_key = '_sku' AND wp_postmeta.meta_value != '' AND wp_posts.post_parent = %d", $post_id )
```
Comment (mian3010) [inline features/woocommerce/woocommerce.php]: There is a couple of places at the end of this line you are directly referencing table names. Should be swapped out with eg. `{$wpdb->postmeta}`

**Comment by mian3010** [comment]
I just took a look at this, and have implemented it in my own custom plugin, so i can use it until it gets merged. I have left a comment on the line with the SQL-query, you have just missed two replacements.

Furthermore, i have added indexing of IDs. Might not be useful for end users, but for admins it is surely convenient to be able to search for post IDs.

Good work @lukaspawlik, it seems to work without major issues! Lets hope it gets merged soon :-)

**Comment by allan23** [comment]
@lukaspawlik can you please resolve the merge conflicts for this PR?

**Comment by allan23** [comment]
Closing due to different codebase in 3.0.

**Pull request closed without merging**